com.ibm.streamsx.topology/com.ibm.streamsx.topology.python/namespace-info.spl

```diff
@@ -0,0 +1,160 @@
+
+When a returned tuple has more values than attributes in the SPL output schema then the additional values are ignored.
+
+++ \@spl.pipe
+Decorator to create a pipe SPL operator.
+When a Python function is decorated with `@spl.pipe`
+a pipe SPL operator is created with a single input port
+and single output port.  For each input tuple the function is called,
+and its return value is used to submit zero or more tuples on the output port. 
+
+When the function returns a tuple containing less values than attributes
+in the SPL output schema then the remaining attributes are copied from
+the input tuple if a matching attribute is found, otherwise they are set
+to the SPL default value.
+
+# Examples
+
+Simple `Noop` pipe operator that passes the input SPL tuple onto its output using a variable argument.
+
+    ＠spl.pipe
+    def Noop(*tuple):
+      "Pass the tuple along without any change"
+      return tuple
+
+Simple filter, note that no return statement is equivalent to returning `None`:
+    ＠spl.pipe
+    def SimpleFilter(a,b):
+      "Filter tuples only allowing output if the first attribute is less than the second. Returns the sum of the first two attributes."
+      if (a < b):
+         return a+b,
+
+Demonstration of returning multiple tuples as a list.
+    ＠spl.pipe
+    def ReturnList(a,b,c):
+      "Demonstrate returning a list of values, each value is submitted as an SPL tuple" 
+      return [(a+1,b+1,c+1),(a+2,b+2,c+2),(a+3,b+3,c+3),(a+4,b+4,c+4)]
+
+
+++ \@spl.sink
+Decorator to create a sink SPL operator.
+If the Python function is decorated with `@spl.sink`
+then the operator is a sink operator, with a single
+input port and no output ports.
+For each input tuple the function is called.
+
+++ \@spl.ignore
+Decorator to ignore a Python function.
+If the Python function is decorated with `@spl.ignore`
+then function is ignored by `spl-python-extract.py`.
+
+++ Extracting SPL operators from decorated Python functions
+
+To create operators in a toolkit, execute `python3 spl-python-extract.py -i toolkit-directory`. Any Python module in the toolkit's `opt/python/streams` directory will have its functions converted to SPL operators.
+
+These directories in a toolkit are automatically added to the Python search path during execution of an operator.
+ * `opt/python/streams` - Contains modules that define Python functions that will be created as SPL operators
+ * `opt/python/packages` - Root directory for Python [https://docs.python.org/3.4/tutorial/modules.html#packages|packages] hierarchy.
+ * `opt/python/modules` - Arbitrary [https://docs.python.org/3.4/tutorial/modules.html#modules|modules], not defined as a packages.
+
+A single Python embedded runtime is used by an SPL processing element (PE), thus when multiple operators
+implemented in python are fused into the same PE they share the same runtime. The library and include
+paths to the Python runtime are set from the version of Python used to execute `spl-extract.py`.
```

Comment:
are couple of places in this file refer to `spl-extract.py` instead of `spl-python-extract.py`